Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -3402,4 +3404,36 @@ final class DeclarationTests: ParserTestCase {
       ]
     )
   }
+
+  func testMissingCommaInParameters() {
+    assertParse(
+      "func a(foo: Bar1️⃣ foo2: Bar2) {}",
+      diagnostics: [
+        DiagnosticSpec(
+          message: "expected ',' in parameter",
+          fixIts: ["insert ','"]
+        ),
+      ],
+      fixedSource: "func a(foo: Bar, foo2: Bar2) {}"
+    )
+  }
+
+  func testMissingMultipleCommasInParameters() {
+    assertParse(
+      "func a(foo: Bar1️⃣ foo2: Bar2, foo3: Bar32️⃣ foo4: Bar4) {}",
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "expected ',' in parameter",
+          fixIts: ["insert ','"]
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
+          message: "expected ',' in parameter",
+          fixIts: ["insert ','"]
+        ),
+      ],
+      fixedSource: "func a(foo: Bar, foo2: Bar2, foo3: Bar3, foo4: Bar4) {}"
+    )
```

Comment:
Could you also add tests for missing commas if the next parameter has an attribute or modifier (see `parseFunctionParameterModifier` or something like that)?